src/ui/generic-components/TrafficLightButton.swift

```diff
@@ -1,219 +1,379 @@
 import Cocoa
 import Foundation
 
-class TrafficLightButton: NSButton {
+/// Enumeration for the different types of traffic light buttons.
+enum TrafficLightButtonType {
+    case quit
+    case close
+    case miniaturize
+    case fullscreen
+}
+
+/// A custom button class representing traffic light buttons in a macOS window.
+class TrafficLightButton: NSView {
+    // Tracks whether the mouse is hovering over the button
     var isMouseOver = false
+    // The type of traffic light button (quit, close, miniaturize, fullscreen)
     var type: TrafficLightButtonType!
-    var window_: Window?
+    // Reference to the window that contains this button
+    var targetWindow: Window?
 
+    /// Custom initializer for the button.
+    ///
+    /// - Parameters:
+    ///   - type: The type of the traffic light button.
+    ///   - tooltip: The tooltip text to display when the user hovers over the button.
+    ///   - size: The size (both width and height) of the button.
     init(_ type: TrafficLightButtonType, _ tooltip: String, _ size: CGFloat) {
         super.init(frame: .init(origin: .zero, size: .init(width: size, height: size)))
         self.type = type
-        target = self
-        action = #selector(onClick)
+
+        // Adjust the button size to fit the specified dimensions
         fit(size, size)
+        // Add a tracking area to detect mouse enter and exit events
         addTrackingArea(NSTrackingArea(rect: bounds, options: [.mouseEnteredAndExited, .activeInKeyWindow], owner: self, userInfo: nil))
         toolTip = tooltip
+        wantsLayer = true
     }
 
+    /// Required initializer for decoding the button (not used here).
     required init?(coder: NSCoder) {
         super.init(coder: coder)
     }
 
-    @objc func onClick() {
-        if (type == .fullscreen) {
-            window_?.toggleFullscreen()
-        } else if (type == .miniaturize) {
-            window_?.minDemin()
-        } else if (type == .close) {
-            window_?.close()
-        } else if (type == .quit) {
-            window_?.application.quit()
-        }
-    }
-
+    /// Mouse entered event handler.
     override func mouseEntered(with event: NSEvent) {
         isMouseOver = true
-        setNeedsDisplay()
+        setNeedsDisplay(bounds)
     }
 
+    /// Mouse exited event handler.
```

Comment:
I don't think these comments are the way to go. When we see a method whith `override`, we can use our IDE to display the original method's documentation, very easily:

<img width="565" alt="image" src="https://github.com/lwouis/alt-tab-macos/assets/106195/48c242ca-b63a-4a0c-91e5-eaa2ec68d09e">